src/builder.rs

```diff
@@ -582,8 +589,11 @@ fn partition_by_asset(
     }
 
     if hm.is_empty() {
-        SpendInfo::dummy(AssetBase::native(), rng);
-        hm.insert(AssetBase::native(), (vec![SpendInfo::dummy(AssetBase::native(), rng), SpendInfo::dummy(AssetBase::native(), rng)], vec![]));
+        let dummy_spend = SpendInfo::dummy(AssetBase::native(), rng);
```

Comment:
This is not gonna fly because we need different nullifiers for dummy notes, I think